lib/active_merchant/billing/gateways/ct_payment.rb

```diff
@@ -0,0 +1,297 @@
+                      commit('preAuthorizationWithToken', post)
+                   else
+                      commit('preAuthorization', post)
+                   end
+        send_ack(response.authorization)
+        response
+      end
+
+      def capture(money, authorization, options={})
+        requires!(options, :order_id)
+        post = {}
+        add_invoice(post, money, options)
+        add_money(post, money)
+        add_customer_data(post, options)
+        transaction_number, authorization_number, invoice_number =
+          split_authorization(authorization)
+        post[:OriginalTransactionNumber] = transaction_number
+        post[:OriginalAuthorizationNumber] = authorization_number
+        post[:OriginalInvoiceNumber] = invoice_number
+
+        response = commit('completion', post)
+        send_ack(response.authorization)
+        response
+      end
+
+      def refund(money, authorization, options={})
+        requires!(options, :order_id)
+        post = {}
+        add_invoice(post, money, options)
+        add_money(post, money)
+        add_customer_data(post, options)
+        transaction_number, _, invoice_number = split_authorization(authorization)
+        post[:OriginalTransactionNumber] = transaction_number
+        post[:OriginalInvoiceNumber] = invoice_number
+
+        response = commit('refundWithoutCard', post)
+        send_ack(response.authorization)
+        response
+      end
+
+      def credit(money, payment, options={})
+        post = {}
+        add_terminal_number(post, options)
+        add_money(post, money)
+        add_operator_id(post, options)
+        add_invoice(post, money, options)
+        add_payment(post, payment)
+        add_address(post, payment, options)
+        add_customer_data(post, options)
+
+        response = payment.is_a?(String) ? commit('refundWithToken', post) : commit('refund', post)
+        send_ack(response.authorization)
+        response
+      end
+
+      def void(authorization, options={})
+        post = {}
+        post[:InputType] = 'I'
+        post[:LanguageCode] = 'E'
+        transaction_number, _, invoice_number =
+          split_authorization(authorization)
+        post[:OriginalTransactionNumber] = transaction_number
+        post[:OriginalInvoiceNumber] = invoice_number
+        add_operator_id(post, options)
+        add_customer_data(post, options)
+
+        response = commit('void', post)
+        send_ack(response.authorization)
+        response
+      end
+
+      def verify(credit_card, options={})
+        post = {}
+        add_terminal_number(post, options)
+        add_operator_id(post, options)
+        add_invoice(post,0, options)
+        add_payment(post, credit_card)
+        add_customer_data(post, options)
+
+        response = commit('verifyAccount', post)
+        send_ack(response.authorization)
+        response
+      end
+
+      def store(credit_card, options={})
+        requires!(options, :email)
+        post = {}
```

Comment:
This is 100% your call, but given you immediately initialize this, you could just do

```ruby
post = {
  LanguageCode: 'E',
  Name: credit_card.name.rjust(50, ' ')
  Email: options[:email].rjust(240, ' ')
}
```

which decreases some repetition. That said, this is 100% your call.